### PR TITLE
add org.gabmus.razercommander

### PR DESCRIPTION
**Pull request by GabMus**

### CODE REVIEW

**Comment by TingPing** [comment]
So this was briefly discussed in IRC but you mentioned this depends upon a specific host daemon for the device driver being available. In general we want applications to be usable after installation but how does this application behave when that daemon is missing? It might be acceptable if it gives a useful error (maybe you can point the user to some host packages) but even then this is less than ideal Flatpak usage.

**Comment by GabMus** [comment]
In the appdata it's specified that the user needs to install the openrazer driver stack.
Will add a message that tells the user to install the driver if it's not installed.

**Comment by nedrichards** [comment]
bot, build org.gabmus.razercommander

File: org.gabmus.razercommander.json [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                    "url": "https://github.com/numpy/numpy/releases/download/v1.13.3/numpy-1.13.3.tar.gz",
+                    "sha256": "4c6b4eef790528bebb7ec9590d74cc193868940fe68e4109a91c196df72d8094"
+                }
+            ]
+        },
+        {
+            "name": "python-openrazer",
+            "buildsystem": "simple",
+            "ensure-writable": [
+                "/lib/python3.6/site-packages/easy-install.pth",
```
Comment (nedrichards) [inline org.gabmus.razercommander.json]: Use a * in the path to be a little more robust here. Example: https://github.com/flathub/work.openpaper.Paperwork/blob/742800038615e1b3552f828b327a9275b8db6162/work.openpaper.Paperwork.json#L42

Comment (GabMus) [reply org.gabmus.razercommander.json]: Done! What about those failing checks?

File: org.gabmus.razercommander.json [hunk 2]
```diff
@@ -0,0 +1,100 @@
+{
+    "app-id": "org.gabmus.razercommander",
+    "command": "razercommander",
+    "branch": "master",
```
Comment (nedrichards) [inline org.gabmus.razercommander.json]: this needs to be 'stable' for flathub, or just left out.

**Comment by nedrichards** [comment]
Failing check is because it's the [wrong branch name](https://github.com/flathub/flathub/wiki/Review-Guidelines).

**Comment by GabMus** [comment]
So I should make a `stable` branch and build the flatpak from there?

**Comment by nedrichards** [comment]
bot, build org.gabmus.razercommander

File: org.gabmus.razercommander.json [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    "command": "razercommander",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.26",
+    "sdk": "org.gnome.Sdk",
+    "command": "razercommander",
+    "finish-args": [
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
+        "--socket=system-bus",
```
Comment (TingPing) [inline org.gabmus.razercommander.json]: This permission is overly broad

File: org.gabmus.razercommander.json [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        "--share=network",
+
+        "--talk-name=org.gnome.SettingsDaemon",
+        "--talk-name=org.razer",
+
+        "--env=DCONF_USER_CONFIG_DIR=.config/dconf",
+        "--filesystem=xdg-run/dconf",
+        "--filesystem=~/.config/dconf:ro",
+        "--filesystem=~/.config/razercommander:rw",
```
Comment (TingPing) [inline org.gabmus.razercommander.json]: This shouldn't be needed.

File: org.gabmus.razercommander.json [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        "--filesystem=~/.config:rw",
```
Comment (TingPing) [inline org.gabmus.razercommander.json]: Overly broad.

Comment (GabMus) [reply org.gabmus.razercommander.json]: I need to create .config/razercommander if it doesnt exist already

File: org.gabmus.razercommander.json (hunk 4 above)
Comment (GabMus) [reply org.gabmus.razercommander.json]: it is, that's where the configuration files are saved

**Comment by GabMus** [comment]
replied to your comments and stripped stuff that isn't necessary

File: org.gabmus.razercommander.json (hunk 4 above)
Comment (TingPing) [reply org.gabmus.razercommander.json]: All applications have a private `$XDG_CONFIG_HOME`, that is where you save configs. Are you saying the daemon also uses that directory and you have to modify it there?

File: org.gabmus.razercommander.json (hunk 5 above)
Comment (TingPing) [reply org.gabmus.razercommander.json]: `--filesystem=~/.config/foo:create` exists for that.

**Comment by nedrichards** [comment]
bot, build org.gabmus.razercommander

File: org.gabmus.razercommander.json (hunk 4 above)
Comment (GabMus) [reply org.gabmus.razercommander.json]: > Are you saying the daemon also uses that directory and you have to modify it there?

No, it doesn't

Alright, so I *can* let razercommander write into its own `$XDG_CONFIG_HOME`, but if I let it use the real `~/.config` folder, the flatpak package will share the same configuration of the regular package. This way users won't have to re-configure profiles and macros for their devices if they migrate to the flatpak version.

This said, I'd prefer to keep this as it is right now, but if using the private `$XDG_CONFIG_HOME` is required to get razerCommander accepted in flathub, I will change it.

File: org.gabmus.razercommander.json (hunk 5 above)
Comment (GabMus) [reply org.gabmus.razercommander.json]: cool, didn't know that. will change that if I can keep on using `~/.config/razercommander`

**Comment by GabMus** [comment]
commented, waiting for replies

File: org.gabmus.razercommander.json (hunk 4 above)
Comment (TingPing) [reply org.gabmus.razercommander.json]: It should be using a private directory, the entire point of flatpak is being self contained.

Comment (GabMus) [reply org.gabmus.razercommander.json]: alright, will change that

**Comment by nedrichards** [comment]
Which changes is this actually awaiting now? Seems 'fine' to me if it has a message to the user about the host daemon requirement if it's absent when starting.

**Comment by TingPing** [comment]
@nedrichards AFAICT that hasn't been added yet, the config dir permissions also haven't been removed yet.

**Comment by barthalion** [comment]
I'm closing it due to inactivity – but please re-open it or resubmit as new pull request if you would like to see it merged in the future.

**Pull request closed without merging**